### ISSUE TITLE
Improve rule documentation

### ISSUE THREAD

**Issue by ganeshrn** [ansible/ansible-lint#1831]
##### Summary

Revisit all the rules from documentation aspects  and update the docs to provide a clear and concise explanation for the rules.
If there are issues open for individual rule docs they can tracked using this ticket. 
##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible-lint --version

```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behaviour

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behaviour

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
Just to give an update. We now have >80% of rules with detailed documentation and looking to achieve 100% before end of the month. Is not that the others do not have docs, but is just very brief.

**Comment by ssbarnea** [comment]
Closing this as done as at this moment all the rules have documentation.
